staking/integration-tests/tests/stability.rs

```diff
@@ -0,0 +1,749 @@
+                    pool_data_pubkey,
+                    stake_account_positions[*delegator],
+                )
+                .unwrap();
+                sanity_check(&mut svm, pool_data_pubkey, &stake_account_positions);
+                operation_counts.get_mut(&operation.get_name()).unwrap().0 += 1;
+            }
+            Operation::CreateGovernancePosition { amount, delegator } => {
+                create_position(
+                    &mut svm,
+                    &props.delegators[*delegator].0,
+                    stake_account_positions[*delegator],
+                    TargetWithParameters::Voting {},
+                    None,
+                    *amount,
+                );
+                operation_counts.get_mut(&operation.get_name()).unwrap().0 += 1;
+            }
+            Operation::CloseGovernancePosition { delegator, amount } => {
+                let mut stake_positions_account =
+                    fetch_positions_account(&mut svm, &stake_account_positions[*delegator]);
+                let positions = stake_positions_account.to_dynamic_position_array();
+
+                let mut index: u8 = 0;
+                let mut position_value = 0;
+                for i in 0..positions.get_position_capacity() {
+                    if let Some(position) = positions.read_position(i).unwrap() {
+                        let position_state = position
+                            .get_current_position(get_current_epoch(&mut svm), UNLOCKING_DURATION)
+                            .unwrap();
+                        if matches!(
+                            position,
+                            Position {
+                                target_with_parameters: TargetWithParameters::Voting { .. },
+                                ..
+                            }
+                        ) && (position_state == PositionState::LOCKED
+                            || position_state == PositionState::LOCKING)
+                        {
+                            index = i.try_into().unwrap();
+                            position_value = position.amount;
+                            break;
+                        }
+                    }
+                }
+
+                if position_value == 0 {
+                    continue;
+                }
+
+                close_position(
+                    &mut svm,
+                    &props.delegators[*delegator].0,
+                    stake_account_positions[*delegator],
+                    TargetWithParameters::Voting {},
+                    None,
+                    min(*amount, position_value),
+                    index,
+                )
+                .unwrap();
+
+                operation_counts.get_mut(&operation.get_name()).unwrap().0 += 1;
+            }
+        }
+    }
+
+    println!("Operation counts:");
+    for (name, (count, total)) in operation_counts.iter() {
+        println!("{}: {}/{}", name, count, total);
```

Comment:
is it expected that here count and total always are equal

Reply:
no the count is the number of operations that succeeded while total is the total number of operations

Reply:
Only undelegate and close position can fail right?

Reply:
delegate and can also fail if there not enough tokens in their stake account